### DISCUSSION TITLE
SMTP_TRANSPORT_JSON Issue with Nodemailer and Port Configuration in Dockerized EnvKey Setup

### DISCUSSION THREAD

**Discussion by mario-lemes** [envkey/envkey#69] [Q&A] [answered]
Description:

I am experiencing an issue with the SMTP_TRANSPORT_JSON configuration and Nodemailer within a Docker container deployment of EnvKey, using the distroless image. This problem is preventing the sending of user signup verification tokens, which is a critical blocker to proceed with user onboarding.

The SMTP_TRANSPORT_JSON environment variable is set as follows:

`SMTP_TRANSPORT_JSON={"host": "xxx", "port": 25, "auth": {"user": "xxx", "pass": "xxx" }}`

According to this configuration, Nodemailer should use port 25 for SMTP transport. However, it appears that Nodemailer is erroneously using port 443, leading to a "connection refused" error.

![image](https://github.com/envkey/envkey/assets/2939658/81eaa3f0-1db6-416e-8b31-5c03cbed1f59)

I followed the deployment instructions from [here](https://docs-v2.envkey.com/docs/self-hosting-open-source-envkey) for community self-hosting EnvKey in a Docker container, but this SMTP/Nodemailer issue has stymied progress.

I would appreciate any insights or potential solutions to resolve this problem. Thank you.

**Comment by danenania** [comment]
Hi @mario-lemes this doesn't look like an SMTP issue but rather that the EnvKey client can't make a connection to your server.

Please make sure this step from the docs work for you:

> 12.) Make a GET request to https://your-subdomain.your-domain.com and confirm you get an API OK response.

**Reply by mario-lemes** [reply]
Hi @danenania,

Thanks for your quick response. 

I'm not sure if that's the case, as you can see I'm able to get 200 OK by calling directly to the subdomain:

![image](https://github.com/envkey/envkey/assets/2939658/dd67a1e0-c2c8-46e7-b17e-0ebf28b14aad)

**Reply by danenania** [reply]
@mario-lemes It looks like you're making an http request rather than an https request--please make sure an https request works too.

**Reply by mario-lemes** [reply]
@danenania I haven't enabled tls since I'm hosting the docker server on a VPN. This VPN has acces to internet via specific maping of IP:port.

I'm able to reach the server with no problem, the issue is on this part, at the Nodemailer transporter creation. https://github.com/envkey/envkey/blob/15157a25815222beed9446049b96246208ef3031/public/app/api/shared/src/email.ts#L76

For some reason is overriding the port passed throug env variable to the port 443.

**Reply by danenania** [reply]
Ok @mario-lemes, can you share the server logs then? In the client logs you sent, the issue is with connecting to the EnvKey server, not SMTP.

**Reply by danenania** [reply]
The EnvKey client can only make https requests, so it's not going to work until you get TLS set up.

**Reply by mario-lemes** [reply]
@danenania Oh that make sense then, is it possible to use self signed certificates?

**Reply by danenania** [reply, marked as answer]
Yes it's possible, though requires starting the core manually inline and passing either `NODE_EXTRA_CA_CERTS=your-cert-file` for validation or else setting `NODE_TLS_REJECT_UNAUTHORIZED=1` (not recommended as it fully disables cert validation--use at your own risk).

So you'd start the core inline with:

```
NODE_EXTRA_CA_CERTS=... envkey core start --inline
``` 

Then open the EnvKey UI. You'll also have to watch out for the UI automatically restarting its own core process if your inline core process stops for any reason--if that happens it won't be run with the env var and cert validation will start failing.

**Reply by mario-lemes** [reply]
Thanks for your help @danenania. That worked, really appreciated the time you spent going through my problem. Have a nice day ahead!

**Reply by danenania** [reply]
That's great to hear @mario-lemes! Same to you!
